File: flutter_app/lib/models/movie.dart
class MovieFile {
  final String quality;
  final String size;
  final String movieId;

  MovieFile({
    required this.quality,
    required this.size,
    required this.movieId,
  });

  factory MovieFile.fromJson(Map<String, dynamic> json) {
    return MovieFile(
      quality: json['quality'],
      size: json['size'],
      movieId: json['movie_id'],
    );
  }
}

class Movie {
  final String title;
  final String imdbID;
  final int? year;
  final List<MovieFile> files;
  String? posterUrl;
  String? rating;
  String? genre;
  String? plot;

  Movie({
    required this.title,
    required this.imdbID,
    this.year,
    required this.files,
    this.posterUrl,
    this.rating,
    this.genre,
    this.plot,
  });

  factory Movie.fromJson(Map<String, dynamic> json) {
    return Movie(
      title: json['title'],
      imdbID: json['imdbID'],
      year: json['year'],
      files: (json['files'] as List)
          .map((f) => MovieFile.fromJson(f))
          .toList(),
    );
  }
}

File: backend/models/movie.py
from pydantic import BaseModel
from typing import List, Optional

class MovieFile(BaseModel):
    quality: str
    size: str
    movie_id: str # internal ID or file_id
    telegram_link: Optional[str] = None
    caption: Optional[str] = None
    file_name: Optional[str] = None
    year: Optional[int] = None
    language: Optional[str] = None
    season: Optional[int] = None
    episode: Optional[int] = None

class Movie(BaseModel):
    title: str
    imdbID: str
    year: Optional[int] = None
    files: List[MovieFile]

class LinkResponse(BaseModel):
    url: str
